mlflow/store/cli.py

```diff
@@ -103,3 +103,14 @@ def download_artifacts(run_id, artifact_path):
     artifact_repo = get_artifact_repository(artifact_uri)
     artifact_location = artifact_repo.download_artifacts(artifact_path)
     print(artifact_location)
+
+
+@commands.command("download-from-uri")
+@click.option("--artifact-uri", "-a", required=True,
+              help="URI pointing to the artifact file or artifacts directory.")
+def download_artifacts_from_uri(artifact_uri):
```

Comment:
What is the difference between this method and https://github.com/mlflow/mlflow/blob/eb51e79d190afb8d0dd5f987cd4be00b4dd744d2/mlflow/data.py#L85, which is already exposed as a CLI command via `mlflow download`?

Also, how does this relate to `mlflow artifacts download`, which should probably be repurposed to operate on absolute URIs (either restricted to URIs with the `runs:/` scheme or operating on the full set of URIs supported by the MLflow artifact repo implementation)

Reply:
`mlflow artifacts download` requires the `run_id` and seems to be part of a set of [CLI commands](https://github.com/mlflow/mlflow/blob/eb51e79d190afb8d0dd5f987cd4be00b4dd744d2/mlflow/store/cli.py#L90) that also require `run_id`. I assumed those were meant to be used within the context of a tracking server. 

`data.download` is what we'd like to use ideally, and it'd make sense for it to do basically what `_download_artifacts_from_uri` does. @andrewmchen was looking at this but we didn't want to block on it for the runs:/ URI work as it may not make it into 1.0. (It's not completely straight-forward because there is a small difference between its behavior and that of `_download_artifacts_from_uri` (in appending the basename to the destination path), and we need to make sure projects don't break after the change).

in summary, to use either of those would require behavior changes for them, so i'd vote for not blocking on that and using the new CLI command, and marking the new command as experimental so we can swap it out as soon as we can with hopefully `mlflow download`.

Reply:
As discussed offline, I'm okay with adding this additional CLI method, pending follow-up PRs to remove the `mlflow download` CLI and, *potentially*, merge this with `mlflow artifacts download`.